src/main/java/com/code_intelligence/jazzer/mutation/mutator/lang/PrimitiveArrayMutatorFactory.java

```diff
@@ -250,6 +254,29 @@ private static AnnotatedType convertWithLength(AnnotatedType type, AnnotatedType
       }
     }
 
+    // The strings we pass to native callbacks to trace data flow are CESU-8 encoded.
+    // As a result, libFuzzer's TORC contains CESU-8 encoded strings.
+    // Therefore, in 50% of times we decode the byte array as a CESU-8  string.
+    public char[] postMutateChars(byte[] bytes, PseudoRandom prng) {
+      if (prng.choice()) {
+        return (char[]) toPrimitive.apply(bytes);
+      } else {
+        char[] chars = new String(bytes, Charset.forName("CESU-8")).toCharArray();
```

Comment:
> This doesn't seem to respect tha length provided by `@WithLength` annotation. 

Oh, it does, never mind!

Reply:
@oetr The length annotation is enforced by the `innerMutator`, which is used by `toPrimitive` and `toPrimitiveAfterMutate`: see https://github.com/CodeIntelligenceTesting/jazzer/blob/CIF-1863-string-compares-on-char-arrays/src/main/java/com/code_intelligence/jazzer/mutation/mutator/lang/PrimitiveArrayMutatorFactory.java#L95

Reply:
Nice catch @Marcono1234 , thanks!